pytensor/tensor/subtensor.py

```diff
@@ -3021,8 +3021,57 @@ def _get_vector_length_Subtensor(op, var):
         raise ValueError(f"Length of {var} cannot be determined")
 
 
+def slice_at_axis(sl: slice, axis: int) -> tuple[slice, ...]:
+    """
+    Construct tuple of slices to slice an array in the given dimension.
+
+    Copied from numpy.lib.arraypad._slice_at_axis
+    https://github.com/numpy/numpy/blob/300096d384046eee479b0c7a70f79e308da52bff/numpy/lib/_arraypad_impl.py#L33
+
+    Parameters
+    ----------
+    sl : slice
+        The slice for the given dimension.
+    axis : int
+        The axis to which `sl` is applied. All other dimensions are left
+        "unsliced".
+
+    Returns
+    -------
+    sl : tuple of slices
+        A tuple with slices matching `shape` in length.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        import pytensor.tensor as pt
+
+        s = pt.slice_at_axis(slice(None, 1), 1)
+        s
+        # Output: (slice(None, None, None), slice(None, 3, -1), (...,))
+
+        x = pt.tensor('x', shape=(None, None, None))
+        x_sliced = x[s]
+
+        f = pytensor.function([x], x_sliced)
+        x = np.arange(27).reshape(3, 3, 3)
+        f(x)
+        # Output: array([[[ 0.,  1.,  2.]],
+        #                [[ 9., 10., 11.]],
+        #                [[18., 19., 20.]]])
```

Comment:
You can make this a test-code block now, so it gets tested by the CI!  That ensures our examples actually work.

https://github.com/pymc-devs/pytensor/blob/4c78408660cefb2066d29d718ed92d04939b7bdb/pytensor/gradient.py#L2080-L2111